parser/src/ast/transition_constraints.rs

```diff
@@ -33,6 +33,8 @@ impl TransitionConstraint {
 pub enum TransitionExpr {
     Const(u64),
     Var(Identifier),
+    VecElem(Identifier, usize),
+    MatrixElem(Identifier, usize, usize),
```

Comment:
Tbh, I'm not 100% convinced by the naming here, since a vector access may not be an element - it might be another vector. Do you have other ideas for this? Lookup, access, index, ... (I'm not in love with anything I've come up with)

Reply:
I think "access" is the best of these, but I agree that it is not satisfying.

Reply:
I'm not sure if it's any better but other options probably are entity or item?